testing/sdk_auth.py

```diff
@@ -282,13 +287,35 @@ def create_remote_keytab(self, name: str, principals: list=[]) -> str:
                                             "runs/latest", name)
         return keytab_absolute_path
 
+
+    @retrying.retry(stop_max_attempt_number=6, wait_fixed=60000)
     def get_keytab_for_principals(self, principals: list, output_filename: str):
         """
         Download a generated keytab for the specified list of principals
         """
         remote_keytab_path = self.create_remote_keytab(self.keytab_file_name, principals=principals)
         _copy_file_to_localhost(self.kdc_host_id, remote_keytab_path, output_filename)
 
+        # In a fun twist, at least in the HDFS tests, we sometimes wind up with a _bad_ keytab. I know, right?
+        # We can validate if it is good or bad by checking it for a certain pattern. That pattern, is when
+        # strings is run against it (remember, strings will show any 4+ consecutive ASCII characters by default)
+        # that some lines will start with ZX.
+        #
+        # See HDFS-493 if you'd like to learn more. Personally, I'd like to forget about this.
+        command = "strings {} | grep ^ZX".format(output_filename)
+        result = subprocess.run(command,
+               shell=True,
+               stdout=subprocess.PIPE,
+               stderr=subprocess.PIPE)
+
+        # Grep exits 0 if there were matches
+        if result.returncode == 0:
+            log.info("There were some matches when checking the keytab for ^ZX: %s", result.stdout)
+            raise Exception("The keytab is bad :(. We're going to retry generating this keytab. What fun.")
```

Comment:
Could maybe relaunch the kdc via marathon and just re-init from scratch. I assume it doesn't have any persistent volumes...

Reply:
It does not, I didn't want to figure out how to unwind things that far, but I want to refactor this majorly anyways.

I'm at least going to incorporate my java checking tool instead of using strings.